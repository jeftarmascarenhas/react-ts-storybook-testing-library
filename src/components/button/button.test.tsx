import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Button from "components/button";

test("Button correctly rendering", () => {
  const { asFragment } = render(<Button />);
  expect(asFragment()).toMatchSnapshot();
});

test("Button show label when label passed", () => {
  const { getByText } = render(<Button label="Button Login" />);
  expect(getByText(/Login/i)).toHaveTextContent("Button Login");
});

test("Button call handle click when user to click with aria title", () => {
  const onClick = jest.fn();
  const { getByTitle } = render(
    <Button
      onClick={onClick}
      label="Facebook Connect"
      title="Facebook button"
    />
  );
  fireEvent.click(getByTitle(/button/i));
  expect(onClick).toHaveBeenCalled();
});
