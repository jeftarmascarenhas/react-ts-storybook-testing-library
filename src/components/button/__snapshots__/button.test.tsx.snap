// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button correctly rendering 1`] = `
<DocumentFragment>
  .c0 {
  border: 1px solid #00be;
  color: 1px solid #00be;
  background-color: 1px solid #00be;
}

.c0:disabled {
  border: 1px solid #ccc;
}

<button
    class="c0"
    color="primary"
    title="button"
  >
    <div
      class=""
    />
  </button>
</DocumentFragment>
`;
